.github/workflows/pull_request_checks.yml

```diff
@@ -0,0 +1,67 @@
+name: Pull Request Checks
+
+on:
+  pull_request:
+    types:
+      - opened
+      
+     # This `edited` flag is why we need a separate workflow - 
+     # specifying edited here causes this job to be re-run whenever someone edits the PR title/description. 
+     
+     # If we had the debug builds in this file, they would run unnecessarily, costing resources.
+      - edited
+
+      - synchronize
+
+permissions:
```

Comment:
I believe the permissions can be per-job, as the example shows.

Does "Pull Request Name Check" modify the PR? If not, make it read-only.